### DISCUSSION TITLE
action fails to fetch artifacts if triggered by pull request, works fine on commit

### DISCUSSION THREAD

**Discussion by mattvenn** [community/community#28868] [Actions] [answered]
I have a Python script in my repo that fetches artifacts from other repositories. This is run from a github action.
The action completes with no problems when I run it manually or it gets triggered by a commit.

However, when someone opens a PR, which triggers the action, the job fails because the Python script can no longer download the artifact. I don't think this is a permission issue because I faced them before and the script always failed with 401. This time it is able to download an artifact, but then the file isn't a zip file.

https://github.com/mattvenn/tinytapeout-mpw7/runs/7742638406?check_suite_focus=true

Any ideas?

**Comment by airtower-luna** [comment, marked as answer]
It's a PR from a fork. Workflows for `pull_request` events from forks [don't get access to secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow), because otherwise anyone could send you a PR with a modified or added workflow and steal or abuse your secrets.

See [Keeping your GitHub Actions and workflows secure Part 1: Preventing pwn requests](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/) for a discussion of the problem and alternatives. And don't run/build untrusted code with access your secrets. :wink:

**Reply by mattvenn** [reply]
Thanks! Makes sense.
